cypress/integration/core-opensearch-dashboards/opensearch-dashboards/apps/query_enhancements/helpers.js

```diff
@@ -0,0 +1,342 @@
+ * @param {string} expectedQueryHitsAfterFilterApplied expected number of hits in string after the filter is applied. Note you should add commas when necessary e.g. 9,999
+ * @param {boolean} shouldMatch boolean to determine if same rowNumber text should match after filter is applied
+ * @example verifyDocTableFilterAction(0, 'filterForValue', '10,000', '1', true)
+ */
+export function verifyDocTableFilterAction(
+  rowNumber,
+  filterElement,
+  expectedQueryHitsWithoutFilter,
+  expectedQueryHitsAfterFilterApplied,
+  shouldMatch
+) {
+  getDocTableField(0, rowNumber).then(($field) => {
+    const shouldText = shouldMatch ? 'have.text' : 'not.have.text';
+
+    const filterFieldText = $field.find('span span').text();
+    $field.find(`[data-test-subj="${filterElement}"]`).click();
+    // Verify pill text
+    cy.getElementByTestId('globalFilterLabelValue', {
+      timeout: 10000,
+    }).should('have.text', filterFieldText);
+    cy.getElementByTestId('discoverQueryHits').should(
+      'have.text',
+      expectedQueryHitsAfterFilterApplied
+    ); // checkQueryHitText must be in front of checking first line text to give time for DocTable to update.
+    getDocTableField(0, rowNumber).find('span span').should(shouldText, filterFieldText);
+  });
+  cy.getElementByTestId('globalFilterBar').find('[aria-label="Delete"]').click();
+  cy.getElementByTestId('discoverQueryHits').should('have.text', expectedQueryHitsWithoutFilter);
+}
+
+/**
+ * Check if the first expanded Doc Table Field's first row's Filter For, Filter Out and Exists Filter buttons are disabled.
+ * @param {boolean} isEnabled Boolean determining if these buttons are disabled
+ */
+export function verifyDocTableFirstExpandedFieldFirstRowFilterForFilterOutExistsFilterButtons(
+  isEnabled
+) {
+  const shouldText = isEnabled ? 'be.enabled' : 'be.disabled';
+  getExpandedDocRow(0, 0).within(() => {
+    cy.getElementByTestId('addInclusiveFilterButton').should(shouldText);
+    cy.getElementByTestId('removeInclusiveFilterButton').should(shouldText);
+    cy.getElementByTestId('addExistsFilterButton').should(shouldText);
+  });
+}
+
+/**
+ * Check the Filter For button in the expandedDocumentRowNumberth field in the expanded Document filters the correct value.
+ * @param {number} docTableRowNumber Integer starts from 0 for the first row
+ * @param {number} expandedDocumentRowNumber Integer starts from 0 for the first row
+ * @param {string} expectedQueryHitsWithoutFilter expected number of hits in string after the filter is removed Note you should add commas when necessary e.g. 9,999
+ * @param {string} expectedQueryHitsAfterFilterApplied expected number of hits in string after the filter is applied. Note you should add commas when necessary e.g. 9,999
+ */
+export function verifyDocTableFirstExpandedFieldFirstRowFilterForButtonFiltersCorrectField(
```

Comment:
can we combine this filter for and filter out  into one? and simplify toggle (exists) one to see if we combine all three?

Reply:
Ok I have combined for and out into one. The toggle is very different so I'm not combining it.